### PR TITLE
add speed test

### PR DESCRIPTION
**Pull request by emlynjdavies**
basic timing of silcam_process without multiprocessing with assertion based on a reasonable computer.

### CODE REVIEW

File: pysilcam/tests/test_speed.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        assert td < pd.to_timedelta('00:00:45'), 'Processing time too long.'
+
+
+
+
+
+
+
+
+
```
Comment (tankefugl) [inline pysilcam/tests/test_speed.py:60]: W391: blank line at end of file

File: pysilcam/tests/test_speed.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import glob
+import os
+import unittest
+import pandas as pd
+import tempfile
+from pysilcam.__main__ import silcam_process
+from pysilcam.silcreport import silcam_report
+from pysilcam.config import load_config
+from pysilcam.postprocess import count_images_in_stats
+from pysilcam.config import PySilcamSettings
```
Comment (tankefugl) [inline pysilcam/tests/test_speed.py:10]: F401: 'pysilcam.config.PySilcamSettings' imported but unused

File: pysilcam/tests/test_speed.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import glob
+import os
+import unittest
+import pandas as pd
+import tempfile
+from pysilcam.__main__ import silcam_process
+from pysilcam.silcreport import silcam_report
+from pysilcam.config import load_config
+from pysilcam.postprocess import count_images_in_stats
```
Comment (tankefugl) [inline pysilcam/tests/test_speed.py:9]: F401: 'pysilcam.postprocess.count_images_in_stats' imported but unused

File: pysilcam/tests/test_speed.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import glob
+import os
+import unittest
+import pandas as pd
+import tempfile
+from pysilcam.__main__ import silcam_process
+from pysilcam.silcreport import silcam_report
```
Comment (tankefugl) [inline pysilcam/tests/test_speed.py:7]: F401: 'pysilcam.silcreport.silcam_report' imported but unused

File: pysilcam/tests/test_speed.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import glob
```
Comment (tankefugl) [inline pysilcam/tests/test_speed.py:1]: F401: 'glob' imported but unused

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: pysilcam/tests/test_speed.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        assert td < pd.to_timedelta('00:00:45'), 'Processing time too long.'
+
+
+
+
+
+
+
+
+
```
Comment (tankefugl) [inline pysilcam/tests/test_speed.py:56]: W391: blank line at end of file

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

**Review by nepstad — APPROVED** [review]

**Review by nepstad — DISMISSED** [review]

File: pysilcam/tests/test_z_classify.py [hunk 7]
```diff
@@ -65,6 +67,11 @@ def correct_positives(category):
 
         # turn failed count into a success percent
         success = 100 - (failed / len(files)) * 100
+
+        t2 = pd.Timestamp.now()
+        td = t2 - t1
+        assert td < pd.to_timedelta(time_limit), 'Processing time too long.'
```
Comment (nepstad) [inline pysilcam/tests/test_z_classify.py:73]: to_timedelta uses nanoseconds as default time unit, probably not was intended here (pass unit='seconds' or similar to be explicit).

**Review by nepstad — APPROVED** [review]

**Review by WillRNaylor — APPROVED** [review]
Looks good.

**Pull request merged**